sdk/go/common/resource/config/crypt.go

```diff
@@ -36,6 +36,7 @@ type Encrypter interface {
 // Decrypter decrypts encrypted ciphertext to its plaintext representation.
 type Decrypter interface {
 	DecryptValue(ciphertext string) (string, error)
+	BulkDecrypt(ciphertexts []string) (map[string]string, error)
```

Comment:
Nit: I don't think that we need this new method on the `Decrypter` interface. Instead, we could let this live on a separate interface and polyfill it for `Decrypter`s that don't implement it. Something like:

```go
type BulkDecrypter interface {
	BulkDecrypt(ciphertexts []string) (map[string]string, error)
}

func BulkDecrypt(dec Decrypter, ciphertexts []string) (map[string]string, error) {
	if bulk, ok := dec.(BulkDecrypter); ok {
		return bulk.BulkDecrypt(ciphertexts)
	}

	result := make(map[string]string, len(ciphertexts))
 	for _, c := range ciphertexts {
		plaintext, err := dec.DecryptValue(c)
		if err != nil {
			return nil, err
		}
		result[c] = plaintext
 	}
 	return result, nil
}
```

That way, not every `Decrypter` has to implement `BulkDecrypt`--only those for which it provides meaningful additional value.

Reply:
@pgavlin I know that @Frassle is going to be in this part of the code soon - he alluded to the fact about so many decrypters :) So I'd love to be able to unlock this value now and defer this to him when he's there

Reply:
I'm fine with it as is. Pauls already done the work of writing BulkDecrypt for everything and as he alluded to I want to do a cleanup pass over the whole lot anyway so if it makes sense I'll do this change then.

Reply:
Sounds good.